src/components/editor/ImagingStudyAttributes.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+
+import React, { Component } from 'react';
+import { RIEInput } from 'riek';
+import _ from 'lodash';
+
+import type { Instance as InstanceType, Series as SeriesType } from '../../types/Attributes';
+
+import { Code } from './Code';
+import { AttributeTemplates } from '../../templates/Templates';
+
+type InstanceProps = {
+  title: string,
+  sop_class: CodeType,
+  onChange: any
+}
+
+export class Instance extends Component<InstanceProps> {
+
+  render() {
+    let title = this.props.title;
+    let sop_class = this.props.sop_class;
+    return (
+      <div>
+        Title: <RIEInput className='editable-text' value={title} propName="title" change={this.props.onChange('title')} />
+        <br />
+        <br />
+        <b>SOP Class:</b>
+        <br />
+        <Code code={sop_class} system={"DICOM-SOP"} onChange={this.props.onChange('sop_class')} />
+      </div>
+    );
+  }
+
+}
+
+type InstanceListProps = {
+  instances: InstanceType[],
+  onChange: any
+}
+
+export class InstanceList extends Component<InstanceListProps> {
+
+  render() {
+    if (!this.props.instances) {
+      return null;
+    }
+    return (
+      <div>
+        {this.props.instances.map((instance, i) => {
+          return (
+            <div className='section' key={i}>
+              <a className='editable-text delete-button' onClick={() => this.props.onChange(`[${i}]`)({val: {id: null}})}>remove</a>
+              <Instance onChange={this.props.onChange(i)} title={instance.title} sop_class={instance.sop_class}/>
```

Comment:
I was able to get the interface to crash once on this line because somehow `instance` was undefined or null after a series of deleting and adding series/instances/etc.  I was unable to recreate the issue after 30 minutes of attempts, so I don't have any more details about how that might have happened.  Since I can't recreate it, I don't think this is a blocking issue, but maybe somebody who has a better understanding of what is happening under the hood here would know.

Reply:
Yeah that's what I figured, but couldn't find the root cause in the code without being able to replicate the error.  So let's just punt on this and keep an eye out.